src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -67,6 +67,14 @@
                 localStorage.setItem(profilerStorageKey + name, value);
             },
 
+            addEvent = function (element, eventName, callback) {
+                if (document.all) {
```

Comment:
> could you do this check around the function definition instead of doing it at runtime to do it only once

@stof, what do you mean? If I'll be checking for existence of `element.addEventListener` then I can't do this outside the `addEvent` function

>  There's also a drawback, in ie8 the reference to this in the event handler refers to the document object instead of the caller object. So eventHandlers using this should have an alternative to the caller object

@wvdweij , what do you mean? I haven't changed `this` reference inside the callback if you're referring to it. And in JavaScript the `this` can be changed dynamically when needed and it's up to callback itself to preserve it.